src/sections/wallet/assets/table/actions/WalletAssetsTableActions.tsx

```diff
@@ -16,168 +15,206 @@ import { useSetAsFeePayment } from "api/payments"
   const { t } = useTranslation()
   const setFeeAsPayment = useSetAsFeePayment()
   const { account } = useAccount()
 
-  const couldBeSetAsPaymentFee =
-    props.couldBeSetAsPaymentFee && !isEvmAccount(account?.address)
+  const navigate = useNavigate()
+
+  const { visible, observe } = useVisibleElements<HTMLDivElement>()
+
+  const {
+    id,
+    symbol,
+    decimals,
+    couldBeSetAsPaymentFee,
+    tradability: { inTradeRouter, canBuy },
+  } = props.asset
+
+  const enablePaymentFee =
+    couldBeSetAsPaymentFee && !isEvmAccount(account?.address)
 
   const couldWatchMetaMaskAsset =
     isMetaMask(window?.ethereum) &&
     isEvmAccount(account?.address) &&
-    props.symbol !== NATIVE_EVM_ASSET_SYMBOL
+    symbol !== NATIVE_EVM_ASSET_SYMBOL
 
-  const isLargeDesktop = useMedia("(min-width: 1100px)")
+  const onFeePaymentSelect = () =>
+    setFeeAsPayment(id, {
+      onLoading: (
+        <Trans
+          t={t}
+          i18nKey="wallet.assets.table.actions.payment.toast.onLoading"
+          tOptions={{
+            asset: symbol,
+          }}
+        >
+          <span />
+          <span className="highlight" />
+        </Trans>
+      ),
+      onSuccess: (
+        <Trans
+          t={t}
+          i18nKey="wallet.assets.table.actions.payment.toast.onSuccess"
+          tOptions={{
+            asset: symbol,
+          }}
+        >
+          <span />
+          <span className="highlight" />
+        </Trans>
+      ),
+      onError: (
+        <Trans
+          t={t}
+          i18nKey="wallet.assets.table.actions.payment.toast.onLoading"
+          tOptions={{
+            asset: symbol,
+          }}
+        >
+          <span />
+          <span className="highlight" />
+        </Trans>
+      ),
+    })
+
+  const buttons: TDropdownItem[] = [
+    {
+      key: "trade",
+      icon: <TradeIcon />,
+      label: t("wallet.assets.table.actions.trade"),
+      onSelect: () =>
+        inTradeRouter
+          ? () =>
+              navigate({
+                to: "/trade/swap",
+                search: canBuy ? { assetOut: id } : { assetIn: id },
+              })
+          : undefined,
+      disabled: !inTradeRouter || account?.isExternalWalletConnected,
+    },
+    {
+      key: "transfer",
+      icon: <TransferIcon />,
+      label: t("wallet.assets.table.actions.transfer"),
+      onSelect: () => props.onTransferClick(),
+      disabled: account?.isExternalWalletConnected,
+    },
+    {
+      key: "deposit",
+      icon: <PlusIcon />,
+      label: t("wallet.assets.table.actions.deposit"),
+      onSelect: () => navigate({ to: LINKS.cross_chain }),
+      disabled: account?.isExternalWalletConnected,
+    },
+  ]
+
+  const hiddenBtnKeys = useMemo(() => {
+    return Object.entries(visible).reduce<string[]>(
+      (memo, [item, isVisible]) => {
+        if (!isVisible) memo.push(item)
+
+        return memo
+      },
+      [],
+    )
+  }, [visible])
```

Comment:
Maybe we can move this logic to useVisibleElements hook and return it as `invisible`